pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -82,24 +70,20 @@ func GetBaseDomain() (string, error) {
 }
 
 // GetPublicZone returns a public route53 zone that matches the name.
-func GetPublicZone(sess *session.Session, name string) (*route53.HostedZone, error) {
-	var res *route53.HostedZone
-	f := func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
-		for idx, zone := range resp.HostedZones {
-			if zone.Config != nil && !aws.BoolValue(zone.Config.PrivateZone) && strings.TrimSuffix(aws.StringValue(zone.Name), ".") == strings.TrimSuffix(name, ".") {
-				res = resp.HostedZones[idx]
-				return false
+func GetPublicZone(ctx context.Context, client *route53.Client, name string) (*route53types.HostedZone, error) {
+	paginator := route53.NewListHostedZonesPaginator(client, &route53.ListHostedZonesInput{})
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list hosted zones: %w", err)
+		}
+
+		for _, zone := range page.HostedZones {
+			if zone.Config != nil && !zone.Config.PrivateZone && strings.TrimSuffix(aws.ToString(zone.Name), ".") == strings.TrimSuffix(name, ".") {
```

Comment:
Do we know if name and zone.name will always be in the form "example.base.domain." or is it possible one comes in as "example.base.domain"?

Reply:
I think:
- `name` input comes from `installconfig.basedomain` so likely it does not have the trailing `.`. `
- `zone.Name`  does have the trailing `.` (i.e. returned from the Routre53 API).

Though, here , we trim it if there is one before comparing anyways :D